test/reservation_manager_test.rb

```diff
@@ -0,0 +1,228 @@
+require_relative 'test_helper'
+
+describe "Reservation Manager Class" do 
+	before do 
+		@sample = HotelManager::ReservationManager.new()
+	end
+
+	it "is an instance of Reservation Manager" do
+		expect(@sample).must_be_kind_of HotelManager::ReservationManager
+		expect(@sample.rooms).must_be_kind_of Array
+		expect(@sample.reservation_blocks).must_be_kind_of Array
+	end
+
+	it "confirm that hotel contains 20 rooms" do
+		expect(@sample.rooms.length).must_equal 20
+	end
+
+	it "able to list out all rooms in the hotel" do
+		expect(@sample.rooms_list()).must_be_kind_of String
+		expect(@sample.rooms_list()).must_include "Room: 1, Room: 2, Room: 3"
+		expect(@sample.rooms_list()).must_include "Room: 19, Room: 20"
+	end
+end
+
+describe "Reservation Manager Class - Allow Booking" do
+	before do 
+		@sample = HotelManager::ReservationManager.new()
+		@sample.save_reservation(Date.new(2020,3,1),Date.new(2020,3,15),num_of_rooms:5)
+	end
+
+	it "allow booking overlapping last day" do
+		@sample.save_reservation(Date.new(2020,3,15),Date.new(2020,3,20),num_of_rooms: 5)
+		expect(@sample.reservation_blocks[1]).must_be_kind_of HotelManager::Reservation
+		expect(@sample.reservation_blocks.length).must_equal 2
+	end
+
+	it "allow individual booking on same range" do
+		@sample.save_reservation(Date.new(2020,3,1),Date.new(2020,3,15))
+		expect(@sample.reservation_blocks[1]).must_be_kind_of HotelManager::Reservation
+		expect(@sample.reservation_blocks.length).must_equal 2
+	end
+
+	it "allow booking within range" do
+		@sample.save_reservation(Date.new(2020,3,2),Date.new(2020,3,14))
+		expect(@sample.reservation_blocks[1]).must_be_kind_of HotelManager::Reservation
+		expect(@sample.reservation_blocks.length).must_equal 2
+	end
+
+	it "allow booking outside of range" do
+		@sample.save_reservation(Date.new(2020,2,25),Date.new(2020,3,18))
+		expect(@sample.reservation_blocks[1]).must_be_kind_of HotelManager::Reservation
+		expect(@sample.reservation_blocks.length).must_equal 2
+	end
+
+	it "allow booking before range" do
+		@sample.save_reservation(Date.new(2020,2,25),Date.new(2020,3,5))
+		expect(@sample.reservation_blocks[1]).must_be_kind_of HotelManager::Reservation
+		expect(@sample.reservation_blocks.length).must_equal 2
+	end
+
+	it "allow booking after range" do
+		@sample.save_reservation(Date.new(2020,3,8),Date.new(2020,3,20))
+		expect(@sample.reservation_blocks[1]).must_be_kind_of HotelManager::Reservation
+		expect(@sample.reservation_blocks.length).must_equal 2
+	end
+end
+
+describe "Reservation Manager Class - Double Booking" do
```

Comment:
I like you test design choice to separate out positive and negative test cases. Consider wrapping these tests in a `save_reservation` describe block since these tests specifically test that method.